src/Data/List/Properties.agda

```diff
@@ -842,6 +842,19 @@ sum-++ (x ∷ xs) ys = begin
 ∈⇒∣product {n} {n ∷ ns} (here  refl) = divides (product ns) (*-comm n (product ns))
 ∈⇒∣product {n} {m ∷ ns} (there n∈ns) = ∣n⇒∣m*n m (∈⇒∣product n∈ns)
 
+nonZero-product : ∀ {ns} → All NonZero ns → NonZero (product ns)
```

Comment:
This proof was described [here](https://github.com/agda/agda-stdlib/pull/1969#discussion_r1435032512) in the course of reviewing @Taneb 's #1969 ...
... and I think is a bit cleaner wrt the use of instance inference?

Reply:
And naming: see above?